Fix `JavaExecGeneratorMainClassDeterminationTest` failing on windows

### Component

JKube Kit

### Task description

### Description
Related to https://github.com/eclipse/jkube/issues/1338

⚠️ A windows machine is required to reproduce and fix problems in this issue

When executing `mvn clean install` on Windows, the following unit tests are failing in `jkube-kit/generator/java-exec` module:

- [JavaExecGeneratorMainClassDeterminationTest](https://github.com/eclipse/jkube/blob/master/jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorMainClassDeterminationTest.java)

It seems to be failing due to this line 
https://github.com/eclipse/jkube/blob/b64915ec6090dcb0171e45b6372052278210e6f3/jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorMainClassDeterminationTest.java#L138

Here we're directly creating a file without specifying parent directory, which creates this file in the directory from where maven build is running, rather than test directory created for the test.

We can fix this test by specifying parent director in File constructor argument like this:
```java
File fatJarArchive = new File(baseDirectory, "fat.jar");
```
Not sure whether to directly create it under base directory or create a nested `target` subdirectory

### Expected Behavior
JavaExecGeneratorMainClassDeterminationTest passes on Windows

### Acceptance Criteria
- [ ] JavaExecGeneratorMainClassDeterminationTest should pass on Windows
- [ ]

I tried running the test again on my windows machine but I run into the same error:
```
[ERROR] Errors:
[ERROR]   JavaExecGeneratorMainClassDeterminationTest.testMainClassDeterminationFromFatJar:154 » IllegalArgument 'other' has different root
[INFO]
[ERROR] Tests run: 18, Failures: 0, Errors: 1, Skipped: 0
[INFO]
```
Could this be because I have this repository cloned in a separate drive?

--- a/jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorMainClassDeterminationTest.java
+++ b/jkube-kit/generator/java-exec/src/test/java/org/eclipse/jkube/generator/javaexec/JavaExecGeneratorMainClassDeterminationTest.java
@@ -135,7 +135,7 @@ class JavaExecGeneratorMainClassDeterminationTest {
     void testMainClassDeterminationFromFatJar(@TempDir File baseDirectory) {
         try (MockedConstruction<FatJarDetector> fatJarDetector = mockConstruction(FatJarDetector.class,
                  (mock, ctx) -> {
-                     File fatJarArchive = new File("fat.jar");
+                     File fatJarArchive = new File(baseDirectory,"fat.jar");
                      when(fatJarDetectorResult.getArchiveFile()).thenReturn(fatJarArchive);
                      when(mock.scan()).thenReturn(fatJarDetectorResult);
                  });

